checkstyle/src/main.rs

```diff
@@ -29,7 +31,9 @@ fn main() -> Result<(), std::io::Error> {
 
     let file_paths = matching_files(&target_path, args.include, args.exclude);
 
-    let nonmatching_files = nonmatching_files(&regex_file_path, file_paths);
+    let nonmatching_files = nonmatching_files(&regex_file_path, file_paths.clone());
+
+    let end_empty_line = no_empty_line(file_paths.clone());
```

Comment:
Let's move this next to the loop we use it in. Also, `end_empty_line = no_empty_line` feels like a semantic error.